### PR TITLE
Fix: Fee token and Account improvements

### PR DESCRIPTION
**Pull request by tabaktoni**
## Development related changes
Restructure fee_token and account

- created Accounts class for dealing with accounts
- cleaned state
- OOP improvement for fee_token extracted from the wrapper
- OOP improvement for account extracted from the wrapper
- wrapper 'starknet' made public accessible 

Not perfect (as starknet_wrapper is now a constructor parameter) but I think it has good direction.
Now both fee_token and account have similar structure and implementation.

## Checklist:

- [x] No linter errors
- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [ ] Linked the issues which this PR resolves
- [ ] Updated the tests
- [x] All tests are passing

### CODE REVIEW

File: starknet_devnet/fee_token.py [hunk 1]
```diff
@@ -32,15 +32,19 @@ class FeeToken:
 
     contract: StarknetContract = None
 
+    def __init__(self, startknet_wrapper):
+        self.startknet_wrapper = startknet_wrapper
```
Comment (FabijanC) [inline starknet_devnet/fee_token.py:36]: typo: startknet

File: starknet_devnet/fee_token.py [hunk 2]
```diff
@@ -69,6 +73,8 @@ async def deploy(self, starknet: Starknet):
             deploy_execution_info=None
         )
 
+        self.startknet_wrapper.contracts.store(FeeToken.ADDRESS, ContractWrapper(self.contract, self.get_contract_class()))
```
Comment (FabijanC) [inline starknet_devnet/fee_token.py:76]: perhaps keep the storage responsibility in starknet wrapper

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: starknet_devnet/starknet_wrapper.py [hunk 3]
```diff
@@ -145,6 +142,10 @@ async def __get_state_root(self):
         state = await self.get_state()
         return state.state.shared_state.contract_states.root
 
+    def store_contract(self, address, contract, contract_class):
```
Comment (FabijanC) [inline starknet_devnet/starknet_wrapper.py:145]: Can you add type hints here? Also, can this method be applied in other places where we do ContractWrapper creation and self.contracts.store

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: starknet_devnet/starknet_wrapper.py (hunk 3 above)
Comment (tabaktoni) [reply starknet_devnet/starknet_wrapper.py]: https://github.com/Shard-Labs/starknet-devnet/pull/203/commits/287f3e3d088c81e832767655c774ee45b303e298

**Review by FabijanC — APPROVED** [review]

**Pull request merged**